### PR TITLE
Adds postgres support matrix support for debian based systems

### PR DESCRIPTION
**Pull request by gurkanindibay**
Build parametrization according to postgres version is handled in docker images for debian based systems since there is only one image for them. Versions should be added into debian/pgversions to be built. Before this PR these parameters are being taken from pkgvars. With this PR , I got these paramaters postgres-versions.txt file.

### CODE REVIEW

File: dockerfiles/ubuntu-xenial-all/scripts/fetch_and_build_deb [hunk 1]
```diff
@@ -49,16 +49,20 @@ fi
 # populate variables from packaging metadata file
 # shellcheck source=/dev/null
 source /buildfiles/pkgvars
-
+source /buildfiles/supported-postgres.txt
```
Comment (hanefi) [inline dockerfiles/ubuntu-xenial-all/scripts/fetch_and_build_deb]: I think we should remove all files for ubuntu/xenial from this repo. Can you do that in a separate PR and update this one accordingly?

File: scripts/fetch_and_build_deb [hunk 2]
```diff
@@ -49,16 +49,20 @@ fi
 # populate variables from packaging metadata file
 # shellcheck source=/dev/null
 source /buildfiles/pkgvars
```
Comment (hanefi) [inline scripts/fetch_and_build_deb:51]: Should we remove this line as well, if we want to stop using `pkgvars`?

File: scripts/fetch_and_build_deb [hunk 3]
```diff
@@ -49,16 +49,20 @@ fi
...
-
+source /buildfiles/supported-postgres.txt
 # set default values for certain packaging variables
 declare pkglatest # to make shellcheck happy
```
Comment (hanefi) [inline scripts/fetch_and_build_deb:58]: What will happen to `pkglatest` when we remove `pkgvars`?

File: scripts/fetch_and_build_deb (hunk 2 above)
Comment (gurkanindibay) [reply scripts/fetch_and_build_deb:51]: getting pkglatest still from pkgvars

File: scripts/fetch_and_build_deb (hunk 3 above)
Comment (gurkanindibay) [reply scripts/fetch_and_build_deb:58]: We will not move pkgvars. We still need to get parameters other than postgres version from pkgvars

File: dockerfiles/ubuntu-xenial-all/scripts/fetch_and_build_deb (hunk 1 above)
Comment (gurkanindibay) [reply dockerfiles/ubuntu-xenial-all/scripts/fetch_and_build_deb]: https://github.com/citusdata/packaging/issues/744

File: scripts/fetch_and_build_deb [hunk 4]
```diff
@@ -49,16 +49,20 @@ fi
 # populate variables from packaging metadata file
 # shellcheck source=/dev/null
 source /buildfiles/pkgvars
-
+source /buildfiles/supported-postgres.txt
```
Comment (hanefi) [inline scripts/fetch_and_build_deb]: I have 2 concerns here:

1. `source` return value is non-zero if the file is not found. This will make the script fail as we have `set -e` option in the file.

```bash
❯ source no-file-with-this-name
source: no such file or directory: no-file-with-this-name
❯ echo $?
127
```

2. We do not need the `.txt` extension. This is Linux territory and looks weird to have extensions.

File: scripts/fetch_and_build_deb [hunk 5]
```diff
@@ -49,16 +49,20 @@ fi
...
 # set default values for certain packaging variables
 declare pkglatest # to make shellcheck happy
 pkgname="${deb_pkgname:-${pkgname}}"
 hubproj="${hubproj:-${pkgname}}"
 nightlyref="${nightlyref:-master}"
-releasepg="${releasepg:-11,12,13}"
-nightlypg="${nightlypg:-${releasepg}}"
+releasepg="${release_versions:-${releasepg}}"
```
Comment (hanefi) [inline scripts/fetch_and_build_deb]: I think your intention is not so easy to read here. You need to explain this better.

Do you want the `release_versions` from to overwrite the value supplied in `releasepg`? Can you do that in a cleaner fashion?

File: scripts/fetch_and_build_deb [hunk 6]
```diff
@@ -49,16 +49,20 @@ fi
...
+nightlypg="${nightly_versions:-${nightlypg}}"
 versioning="${versioning:-simple}"
 
+echo "Postgres versions:"
+echo "Release Versions:${releasepg}"
+echo "Nightly Versions:${nightlypg}"
```
Comment (hanefi) [inline scripts/fetch_and_build_deb]:
```suggestion
echo "Release Versions :${releasepg}"
echo "Nightly Versions :${nightlypg}"
```

**Review by hanefi — CHANGES_REQUESTED** [review]

File: scripts/fetch_and_build_deb (hunk 4 above)
Comment (gurkanindibay) [reply scripts/fetch_and_build_deb]: 1. I see. my intention is not to fail the script in case of file's nonexistence. So I think I need to handle the error here.

File: scripts/fetch_and_build_deb (hunk 5 above)
Comment (gurkanindibay) [reply scripts/fetch_and_build_deb]: My intention is to use pkgvars releasepg values if file not found

File: scripts/fetch_and_build_deb (hunk 4 above)
Comment (gurkanindibay) [reply scripts/fetch_and_build_deb]: Fixed

File: scripts/fetch_and_build_deb (hunk 5 above)
Comment (gurkanindibay) [reply scripts/fetch_and_build_deb]: Changed variable names and added some comments

Comment (gurkanindibay) [reply scripts/fetch_and_build_deb]: After addressing all comments, I will manually create packaging-test images to test the developments made here on https://github.com/citusdata/tools/pull/167

File: scripts/fetch_and_build_deb [hunk 7]
```diff
@@ -49,15 +49,25 @@ fi
 # populate variables from packaging metadata file
 # shellcheck source=/dev/null
 source /buildfiles/pkgvars
+# populate pg versions to be used for packaging.
+# supported-potgres file is derived file from postgres-matrix.yml file
+source /buildfiles/supported-postgres 2>/dev/null
```
Comment (hanefi) [inline scripts/fetch_and_build_deb]: This will fail if the file does not exist. Redirecting stderr does not prevent nonzero vales returned by `source`

Comment (gurkanindibay) [reply scripts/fetch_and_build_deb]: image

Comment (hanefi) [reply scripts/fetch_and_build_deb]: You can check exit code of the last command by `echo $?`

Alternatively, you can use `set -euo pipefail` in bash to have a similar setting on your terminal.

Not having any output on your screen does not guarantee that the command ran successfully.

Comment (gurkanindibay) [reply scripts/fetch_and_build_deb]: You're right I added file existence check now
https://github.com/citusdata/packaging/blob/cc3368bd5400d7c3e7bd0ec59c56b46d1d7741d4/scripts/fetch_and_build_deb#L55

File: scripts/fetch_and_build_deb [hunk 8]
```diff
@@ -49,15 +49,26 @@ fi
...
 versioning="${versioning:-simple}"
+# Fetch pg release and nightly versions from supported-postgres file which is originated from postgres-matrix.yml file
+# If this file is not found, releasepg and nightlypg parameters from pkgvars are used for defining pg versions
+pg_release_versions="${release_versions:-${releasepg}}"
+pg_nightly_versions="${nightly_versions:-${nightlypg}}"
+
+
+echo "Postgres versions:"
+echo "Release Versions :${pg_release_versions}"
+echo "Nightly Versions :${pg_nightly_versions}"
```
Comment (hanefi) [inline scripts/fetch_and_build_deb]:
```suggestion
echo "Release Versions: ${pg_release_versions}"
echo "Nightly Versions: ${pg_nightly_versions}"
```

File: scripts/fetch_and_build_deb [hunk 9]
```diff
@@ -49,15 +49,26 @@ fi
...
 pkgname="${deb_pkgname:-${pkgname}}"
 hubproj="${hubproj:-${pkgname}}"
 nightlyref="${nightlyref:-master}"
-releasepg="${releasepg:-11,12,13}"
-nightlypg="${nightlypg:-${releasepg}}"
 versioning="${versioning:-simple}"
+# Fetch pg release and nightly versions from supported-postgres file which is originated from postgres-matrix.yml file
+# If this file is not found, releasepg and nightlypg parameters from pkgvars are used for defining pg versions
+pg_release_versions="${release_versions:-${releasepg}}"
+pg_nightly_versions="${nightly_versions:-${nightlypg}}"
```
Comment (hanefi) [inline scripts/fetch_and_build_deb:66]:
```suggestion
pg_release_versions="${release_versions:-$releasepg}"
pg_nightly_versions="${nightly_versions:-$nightlypg}"
```

File: scripts/fetch_and_build_deb [hunk 10]
```diff
@@ -49,15 +49,26 @@ fi
 # populate variables from packaging metadata file
 # shellcheck source=/dev/null
 source /buildfiles/pkgvars
+# populate pg versions to be used for packaging.
+# supported-potgres file is derived file from postgres-matrix.yml file
```
Comment (hanefi) [inline scripts/fetch_and_build_deb]:
```suggestion
# Read PostgreSQL versions from file.
# supported-postgres file is derived file from postgres-matrix.yml file by citus_package.
```

**Review by hanefi — APPROVED** [review]
Minor formatting suggestions that should be applied to all the files that are generated from the templates.

File: scripts/fetch_and_build_deb (hunk 9 above)
Comment (gurkanindibay) [reply scripts/fetch_and_build_deb:66]: I think we should keep this since it is better using bash parameters with curly braces
